Documentation/boostrap-new-os.md

```diff
@@ -0,0 +1,140 @@
+So for tools version `2.0.0` on linux distro with x64 architecture where portable dotnet core can be used, the URL is `https://dotnetcli.azureedge.net/dotnet/Sdk/2.0.0/dotnet-sdk-2.0.0-linux-x64.tar.gz`
+To download it, `wget` or `curl` tools can be used.
+Once the file is downloaded, create a new folder and untar the file into it.  
+So e.g. for the file mentioned above, use:
+```bash
+tar -xf dotnet-sdk-2.0.0-linux-x64.tar.gz
+```
+### Choosing RID for the new OS
+The new RID represents your target OS. It is used by developers to target that OS. The RID format is `<Name>.<Version>`. The name is lower case and should match the target OS. The version is optional and should be used if it is expected that in future versions the libraries that .NET Core depends on will not be binary compatible with the first supported version. So for example for FreeBSD 11, the RID would likely be `freebsd.11`.
+For OS that has `/etc/os-release` file, the RID needs to match the `$ID.$VERSION_ID` extracted from that file. 
+### Prerequisites
+The following libraries and tools need to be installed in order to build the bootstrap CLI. The precise names of the packages are OS specific.
+#### Tools
+* Clang version 3.5 or higher, 3.9 recommended
+* CMake version 2.8.12 or higher
+* Python version 2.7
+* AWK
+* SED
+#### Development libraries
+* libunwind
+* lttng-ust
+* openssl or libressl
+* krb5
+* curl
+* liblldb
+* icu
+* libuuid
+### Building the bootstrap CLI
+There is a bash script file that automatizes most of the process of building the bootstrap CLI end to end. It first creates a folder named by the new target RID, clones the coreclr, corefx and core-setup repos into it and checks out the same commit of each of the repos as the one that was used to build the seed CLI. This first step is skipped if the coreclr, corefx and core-setup folders already exist. This is important so that the sources can be modified to fix possible build issues and to target the new RID.
+The script needs to be passed several arguments. The target architecture, the build configuration, the target OS, the new RID and path to the folder with the untared seed CLI. There is also an optional option to specify the version of the Clang compiler to use to compile the native code. There is also an option to pass in a System.Private.CoreLib.dll built elsewhere and an option to crossgen it. This is useful if you are building debug configuration, since the seed CLI that comes from Azure is built for release configuration and the release version of System.Private.CoreLib.dll is not compatible with debug version of libcoreclr.so.
+Here is the summary of the options that you get from running the script with `--help` option:
+```
+Usage: buildbootstrapcli.sh [BuildType] --rid <Rid> --seedcli <SeedCli> [--arch <Architecture>] [--os <OS>] [--clang <Major.Minor>] [--corelib <CoreLib>] [--crossgen]
```

Comment:
`--rid` - the usage of the term "RID" here and throughout is slightly different than what we normally use the term "RID".

`Architecture` is normally included in the RID, and not a separate concept.  Or if it is a separate concept, then they are duplicated.

Maybe we could rename `--rid` to `--base-rid`?  To indicate that this isn't the full RID.

Reply:
I will change that so that the rid is really the full rid and remove the --arch option. Parsing it internally to get the arch is easy.